ballerina/annotation_types.bal

```diff
@@ -45,11 +45,13 @@ public annotation AutoIncrementConfig AutoIncrement on record field;
 # + reference - The names of the fields of the other entity, which are referenced by the `keyColumns`
 # + onDelete - The action to be taken when the referenced value in the parent entity is deleted
 # + onUpdate - The action to be taken when the referenced value in the parent entity is updated
+# + joiningTable - The details of the joining table to be used in a many-to-many relation
 public type RelationConfig record {|
     string[] keyColumns?;
     string[] reference?;
     ReferenceAction onDelete?;
     ReferenceAction onUpdate?;
+    string joiningTable?;
```

Comment:
This would be mandatory as per V2 design. Shall we do the change here itself?

Reply:
This would only be mandatory for many-to-many relations right? For one-to-one and many-to-one it would not be necessary.

In that case, a compiler plugin validation would be needed

Reply:
Ack. Discussed and decided to let this be nillable for now.